apps/frontend/src/modules/quiz/slices/quiz.slice.ts

```diff
@@ -101,17 +128,29 @@ const { actions, name, reducer } = createSlice({
 	name: "quiz",
 	reducers: {
 		nextQuestion(state) {
-			state.currentCategoryIndex += PREVIOUS_INDEX_OFFSET;
-			state.currentCategory =
-				state.questions[state.currentCategoryIndex] || null;
+			if (
+				(state.questionsByCategories as QuizQuestionDto[][]).length >
+				state.currentCategoryIndex
```

Comment:
I think the length will always be greater than index

Reply:
Without this check we get a screen without questions when we try to retake the quiz two times in a row

Reply:
But the category index cannot be equal to the length, because `list[list.length]` is always undefined. The last item is `list[list.length - 1]`. So I think this check will always return `true`